src/CompletionProvider.php

```diff
@@ -121,45 +123,71 @@ public function __construct(DefinitionResolver $definitionResolver, ReadableInde
      */
     public function provideCompletion(PhpDocument $doc, Position $pos): CompletionList
     {
+        // This can be made much more performant if the tree follows specific invariants.
         $node = $doc->getNodeAtPosition($pos);
 
-        if ($node instanceof Node\Expr\Error) {
-            $node = $node->getAttribute('parentNode');
+        $offset = $node === null ? -1 : $pos->toOffset($node->getFileContents());
+        if ($node !== null && $offset > $node->getEndPosition() &&
+            $node->parent->getLastChild() instanceof PhpParser\MissingToken
```

Comment:
You should check whether the node has a parent first to resolve a bug I noticed.